### ISSUE TITLE
Remove redundant signing infra

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/aspnetcore#38739]
We have a bunch of infrastructure for setting up signing in this repo, which is now completely unused in 6.0 & main due to post-build signing (See https://github.com/dotnet/aspnetcore/blob/3d66a59217502e1f0039c0264a5fc2e18c362004/.azure/pipelines/ci.yml#L47-L48). We should be able to remove anything that's conditioned on `PostBuildSign=false`.

Related: https://github.com/dotnet/aspnetcore/pull/26506

**Comment by dougbu** [comment]
Yeah, we've baked for a while w/ post-build signing enabled and been mostly (see #38629) :ok:

➕ @mmitche in case we have some reason to hang onto the old in-build signing stuff.

**Comment by mmitche** [comment]
Let's hang onto it for a tad longer, in case we run into other interesting cases for 6.0 servicing that cause us to reconsider
